Fix various errors when importing stdatomic.h.

This was brought up in the following NDK issues, but is still not working correctly with NDK r22 within extern "C" and with the custom atomic macros in libdispatch:
https://github.com/android/ndk/issues/1177
https://github.com/android/ndk/issues/1178

diff --git a/src/shims/atomic.h b/src/shims/atomic.h
index c002e72..11fc485 100644
--- a/src/shims/atomic.h
+++ b/src/shims/atomic.h
@@ -35,7 +35,11 @@
 #if defined(__cplusplus)
 #define _Bool bool
 #endif
+#if defined(ANDROID) && __has_include(<bits/stdatomic.h>)
+#include <bits/stdatomic.h>
+#else
 #include <stdatomic.h>
+#endif
 
 #define memory_order_ordered    memory_order_seq_cst
 #define memory_order_dependency memory_order_acquire
